### PR TITLE
COM-19920: As an editor, I want to be able to embed code examples

### PR DESCRIPTION
**Pull request by clash82**
https://jira.ez.no/browse/COM-19920

screen shot 2016-09-27 at 17 01 30

### CODE REVIEW

File: app/Resources/views/pagelayout.html.twig [hunk 1]
```diff
@@ -21,6 +22,12 @@
             <link rel="stylesheet" type="text/css" href="{{ asset_url }}">
         {% endstylesheets %}
 
+        {% javascripts
+            'assets/js/prism.js'
```
Comment (SylvainGuittard) [inline app/Resources/views/pagelayout.html.twig]: I would prefer to include this file only on the blog post full view and not for every page of the website.
Possible?

File: src/AppBundle/Twig/FieldOptionsTwigExtension.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * Returns a list of functions to add to the existing list.
+     *
+     * @return array
+     */
+    public function getFunctions()
+    {
+        return [
+            new Twig_SimpleFunction('ez_field_options', [$this, 'showFieldOptions'], [
```
Comment (SylvainGuittard) [inline src/AppBundle/Twig/FieldOptionsTwigExtension.php]: I totally understand this twig function. 
IMHO, this should be included in the product as the other functions (https://doc.ez.no/display/EZP/Twig+functions).
What do you think?

File: app/Resources/views/pagelayout.html.twig (hunk 1 above)
Comment (clash82) [inline app/Resources/views/pagelayout.html.twig]: Nope :-) but in the future I'll create a dedicated block for the assets section in the pagelayout file to load assets depending on the full view.

File: src/AppBundle/Twig/FieldOptionsTwigExtension.php (hunk 2 above)
Comment (clash82) [inline src/AppBundle/Twig/FieldOptionsTwigExtension.php]: Totally agree.

Platform team, what do you think?
/cc @bdunogier @andrerom

**Comment by clash82** [comment]
ping @kamilmusial @damianz5 @SylvainGuittard

File: app/Resources/views/embed/code.html.twig [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{% block content %}
+    {% set languageOptions = ez_field_options('code', 'language') %}
+    {% set language = 'none' %}
+    {% set embedLanguage = ez_field_value(content, 'language') %}
+    {% if languageOptions[embedLanguage] is defined %}
+        {% set language = languageOptions[embedLanguage] %}
```
Comment (kamilmusial) [inline app/Resources/views/embed/code.html.twig]: A little bit too much logic here IMO, have you consider placing it in twig extension? Just a question.

Comment (clash82) [inline app/Resources/views/embed/code.html.twig]: Well... dedicated Twig helper was already added with this PR. Besides it's only one IF condition ;)

**Pull request merged**